src/parser/demonhunter/havoc/modules/spells/BladeDance.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SPECS from 'game/SPECS';
+import Events, { DamageEvent, FightEndEvent } from 'parser/core/Events';
+import { ThresholdStyle, When } from 'parser/core/ParseResults';
+import SpellLink from 'common/SpellLink';
+
+//For Blade dance and Death Sweep
+class BladeDance extends Analyzer {
+  get suggestionThresholds() {
+    return {
+      actual: this.badCast,
+      isGreaterThan: {
+        minor: 0,
+        average: 0,
+        major: 1,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  badCast = 0;
+  hitCount = 0;
+  hitTimeStamp: number = 0;
+  firstHitTimeStamp: number = 0;
+  strikeTime: number = 1000;
```

Comment:
Is this always one second or does it change with haste?

Reply:
I couldnt find a way to calculate it properly with haste, but its max 1 second, and its impossible to cast it again during that time.

Reply:
Anything with haste is just `duration / (1 + %haste)`

Reply:
Yeah ik, but how do I get the correct haste value? I know you can do that in the Abilities script, but I couldnt find out where you actually get the value from.

Saw some spell scripts using functions to get the haste value, but it feels like Im overdoing it for something as this simple.

Reply:
import statTracker and do
statTracker.hastePercentage(statTracker.currentHasteRating));

Reply:
![image](https://user-images.githubusercontent.com/4808216/103505142-d5259d00-4e59-11eb-92d5-ab23ad8a88a1.png)
(last hit - first hit) = total time
373823500-373822541=959

1000 / (1 + 0.17303030303030303) = 852.492895893

According to this its not affected by haste. Looks more like its just max 1 second and it depend on where targets are located(in the hit area) and how many during the last hit, as other strikes have different total times, but no one is matching the haste calculation.

Reply:
Lit